TESTING.md

```diff
@@ -0,0 +1,291 @@
+
+* Dockerfile check:
+  * Does it include nami modules which contain binaries? If so, use the `check-binaries.yaml` GOSS template.
+  * Does it include the ca-certificates package? If so, use the `check-ca-certificates.yaml` template.
+* Compilation logic check:
+  * Are there files or directories created and/or with permission changes? If so, use the `check-directories.yaml` and/or `check-files.yaml` templates.
+  * Are there additional files/dirs modifications? If so, use custom filesystem tests.
+  * Add tests for any compilation options or flags used.
+* Check postunpack script:
+  * Are there files or directories created and/or with permission changes? If so, use the `check-directories.yaml` and/or `check-files.yaml` templates.
+  * Are there additional files/dirs modifications? If so, use custom filesystem tests.
+* Check apps version:
+  * If `$APP_VERSION` follows semver version, use the `check-app-version.yaml` template.
+* Check apps-dependant tests:
+  * If the app is a runtime, test if the runtime can run a compatible file.
+  * If the app requires yet-to-run initialization logic:
+    * No complex configuration nor testing environment is created manually.
+    * If possible, test the app's basic features.
+  * If the app is just a part of a bigger setup (exporter, multi-container apps, etc.):
+    * No testing environment is created manually.
+    * If possible, test the app's basic features.
+  * If the app uses subcomponents (java/php, apache, etc.):
+    * If possible, test a subcomponent when there is a custom config added to them.
+    * Verify whether the subcomponent is capable of running the app.
+* Must-have templates to be added to every suite:
+  * `check-linked.libraries.yaml`
+  * `check-broken-symlinks.yaml`
+  * `check-sed-in-place.yaml`
+  * `check-spdx.yaml`
+* Final checks:
+  * When possible, NO per-branch tests are used.
+  * Every GOSS template is included in `goss.yaml` and the needed vars are in place.
+
+### Specific GOSS acceptance criteria
+
+* [ ] No distro-specific tests are included.
+* [ ] Prioritise using the tests included in the templates over creating custom tests.
+* [ ] Due to GOSS limitations, there can't be two tests with the same name or checking the same file/directory. In those cases, only one of them will be run.
+* [ ] For clarity purposes, the vars needed for the GOSS templates shouldn't be used in the custom tests at `app.yaml`.
+* [ ] Tests should not rely on system packages (e.g. `curl`). Favor built-in GOSS primitives instead.
+* [ ] Prefer checking the exit status of a command rather than looking for a specific output. This will avoid most of the potential flakiness.
+
+### Run GOSS locally
+
+Sometimes it is of interest to run the tests locally, for example during development. Though there may be different approaches, you may follow the steps below to execute the tests locally:
+
+1. Download the [GOSS binary for Linux AMD64](https://github.com/goss-org/goss/releases/)
```

Comment:
```suggestion
1. Download the [GOSS binary for Linux](https://github.com/goss-org/goss/releases/)
```

Reply:
Good catch, I'll also modify the `amd64`reference below.